src/log/Log.cc

```diff
@@ -235,15 +235,15 @@ void Log::_flush_logbuf()
   }
 }
 
-void Log::_flush(EntryVector& t, bool requeue, bool crash)
+void Log::_flush(EntryVector& t, bool crash)
 {
   long len = 0;
+  if (t.empty()) {
+    return;
```

Comment:
this change the behavior of this function.

before this change, `Log::_flush()` always flushes `m_log_buf` to disk. 

after this change, `Log::_flush()` is a no-op as long as `t` is empty.

as you sure this is expected?

Reply:
The `m_log_buf` will be fed by `_flush()` only when `t` is not empty and then immediately it will be flushed to log files and emptied just before returned from `_flush()`.

In other word, `m_log_buf` just a buffer for the log messages in `_flush()` before flushing them to log files.

We should also remove the useless `_flush_logbuf()' in other places which just after the `_log_message()`. IMHO, it maybe supposing that the `_log_message()` will update the `m_log_buf` instead of writing to the `m_fd` directly, but it is not.

```
362   sprintf(buf, "  %2d/%2d (syslog threshold)", m_syslog_log, m_syslog_crash);
363   _log_message(buf, true);
364   sprintf(buf, "  %2d/%2d (stderr threshold)", m_stderr_log, m_stderr_crash);
365   _log_message(buf, true);
366   sprintf(buf, "  max_recent %9zu", m_max_recent); 
367   _log_message(buf, true);
368   sprintf(buf, "  max_new    %9zu", m_max_new);
369   _log_message(buf, true);
370   sprintf(buf, "  log_file %s", m_log_file.c_str());
371   _log_message(buf, true);
372    
373   _log_message("--- end dump of recent events ---", true);
374    
**375   _flush_logbuf();**                                                                                                                                                     
376 
377   m_flush_mutex_holder = 0;
378 }  
```

Reply:
this makes sense. could you revise the commit message to expand "there is no need" a little bit? i think the first paragraph in your reply would suffice.